app/view_objects/allocations_view.rb

```diff
@@ -79,12 +79,14 @@ def build_repeat_allocations(matching_allocation, training_provider)
       allocation_status[:status] = Status::REQUESTED
       allocation_status[:status_colour] = Colour::GREEN
       allocation_status[:requested] = Requested::YES
+      allocation_status[:id] = matching_allocation.id
```

Comment:
This method (and the one below) should have singular method names. I wonder if an `AllocationView` (i mean a singular object that is a view representation of one `Allocation`) would be a better more testable way of implementing this too? May be a bit of a stretch for this PR I realise.